Enterprise-CMCS/macpro-mdct-mfp
# Autosave Bandaid for EntityDetailsOverlay

### Description

This solution is a combination of ideas and test code by @braxex, @ntsummers1 and @ajaitasaini during our huddle. Essentially to fix the delay caused by hydration and component rerendering, we add a spinner to EntityDetailsOverlay return button and wait until autosave is finish before we allow the user to return.


https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/127151429/e586ff15-a59f-42ff-98c6-7c29de1b0d8c



Possible Issue:
- If for some reason, autosave stops midway and doesn't finish, the state can be trapped and the user wouldn't be able to get the return button to work. I haven't been able to trigger it but the possibility is there. If that happens, they still have the save and return button but that does validation checks.

### Related ticket(s)


---
### How to test

1) Sign into MFP
2) Create a new Work Plan
3) In the WP, go to. State and Territory-Specific Initiatives section, create a new Initiative 
4) In the newly create Initiative, go to Define Initiative
5) This change is the most apparent when filling out the checkboxes and not removing the focus
6) Click Return to dashboard for this initiative
    - You should see a spinner only if autosave is running, if there's no autosaving, it'll stay as an arrow

### Important updates




---
### Author checklist


---


_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_
